Services/importHelper.php

```diff
@@ -0,0 +1,232 @@
+            'currentProject' => $_SESSION['csv_data']['project_id'], // Project id is set in importSettings
+        ]);
+
+        $this->projectMilestones = array_map(function ($milestone) {
+            return [
+                'id' => $milestone->id,
+                'headline' => $milestone->headline,
+            ];
+        }, $milestoneData);
+    }
+
+     /**
+     * Check if a given milestone exists in the database for the current project
+     *
+     * @return bool|string
+     *
+     */
+    public function checkMilestoneExist(string $milestone, bool $update = false): bool|string
+    {
+        $milestones = $this->getProjectMilestones($update);
+        return array_reduce($milestones, function ($carry, $item) use ($milestone) {
+            return $carry !== false ? $carry : ($item['headline'] === $milestone ? $item['id'] : false);
+        }, false);
+    }
+
+     /**
+     * Adds a given milestone
+     *
+     * @return array<string, string>|bool|int
+     *
+     */
+    public function addMilestone(string $milestone): array|bool|int
+    {
+        $params = array(
+            'headline' => $milestone,
+        );
+        return $this->ticketService->quickAddMilestone($params);
+    }
+
+/**
+     * Gets all supported fields for mapping
+     *
+     * @return array<string, array<string, string>>
+     *
+     */
+    public function getSupportedFields(): array
+    {
+        $supportedFields = array(
+            'headline' => array(
+                'name' => 'Titel',
+                'help' => '',
+            ),
+            'description' => array(
+                'name' => 'Beskrivelse',
+                'help' => '',
+            ),
+            'tags' => array(
+                'name' => 'Tags',
+                'help' => 'Mapping only supports one tag. Multiple words will become a single tag.',
+            ),
+            'milestoneid' => array(
+                'name' => 'Milestone',
+                'help' => '',
+            ),
+            'planHours' => array(
+                'name' => 'Planlagte timer',
+                'help' => 'Mapping this will also set "Hours left" as the same value to save you time.',
+            ),
+            'dateToFinish' => array(
+                'name' => 'Due Date',
+                'help' => '',
+            ),
+        );
+
+
+        return $supportedFields;
+    }
+
+    /**
+     * Validated the mapped data. Eventually sets warnings and errors in session
+     *
+     * @return void
+     *
+     */
+    public function dataValidationCheck(): void
+    {
+        unset($_SESSION['csv_data']['warnings']);
+        unset($_SESSION['csv_data']['errors']);
+
+        $data = $_SESSION['csv_data']['data'];
+        $params = $_SESSION['csv_data']['mapping_data'];
+        $supportedFields = $this->getSupportedFields();
+
+        // Loop field mapping
+        foreach ($params as $key => $param) {
+            $key = str_replace('_', ' ', $key);
```

Comment:
The array keys in this loop is ultimately user defined, which freaks me out. Php might be able to handle spaces, double spaces etc, but i see this as a safety measure and i feel that it is better practice. 

Do you think otherwise?